### ISSUE TITLE
Different behaviour between Python 2 and 3

### ISSUE THREAD

**Issue by PeterJCLaw** [spulec/freezegun#85]
Hi,

I've been using freezegun to test code which supports both Python 2 and 3, and found that freezegun appears to result in different time values being returned by `datetime.now(timezone)` under different Python versions.

I've cut this down into a gist: https://gist.github.com/PeterJCLaw/96a68c04d0772ccb1182, which contains both the code and the outputs I'm seeing. Those outputs are from version 0.3.1, but I believe I've confirmed that the behaviour is the same using the current master branch as well.

I _think_ based on what's there that it's the Python 3 behaviour which is wrong, but I'm not sure which is expected.

Is this behaving as you'd expect?

Thanks,
Peter

**Comment by spulec** [comment]
After digging into this a bit, I don't think the issue is with freezegun. Try running the following on python2 and python3 and notice the different results.

```python
import datetime
from dateutil.tz import *
tz = gettz('Europe/London')
print(tz.fromutc(datetime.datetime(2014, 4, 26, 12, 1).replace(tzinfo=tz)))
```

Also notice that if we change the datetime to `datetime.datetime(2014, 4, 26, 12, 1)`, then everything works.

I don't have more time to dig into this right now. If you keep digging, let me know what you find.

**Comment by PeterJCLaw** [comment]
Thanks. It turns out that it only expresses an issue when the timezone has an ordinary offset of 0 (ie UTC), but does have a summer-time component.
I've raised an issue against Python 3.4 here: http://bugs.python.org/issue23600.

**Comment by spulec** [comment]
Hmmm, very interesting.

I'm going to close this for now. Feel free to reopen if I can be of assistance or you think there is an easy workaround for freezegun to use, although I am a bit worried it could create more confusion. Also, feel free to comment here if any progress is made on your Python ticket.

**Comment by PeterJCLaw** [comment]
Update: It's been fixed in the python 3.4, 3.5 and 3.6 streams.

**Comment by spulec** [comment]
Awesome, thanks!
